content/api/snippets.apib

```diff
@@ -0,0 +1,211 @@
+Snippets can be imported into any template content or transmission content using substitution syntax.
+
+For example, if you have a snippet with the ID `ourfooter` and the following HTML content:
+
+```html
+<footer>
+    <p>Our HTML footer</p>
+</footer>
+```
+
+The snippet can be imported into an HTML email's content using the `render_snippet` macro call.
+
+```html
+<html>
+    <p>Our body content</p>
+    {{ render_snippet( "ourfooter" ) }}
+</html>
+```
+
+The resulting rendered HTML content would look like:
+
+```html
+<html>
+    <p>Our body content</p>
+    <footer><p>Our HTML footer</p></footer>
+</html>
+```
+
+See the [Substitution Reference](/api/substitutions-reference/#header-snippets)
+for a more detailed explanation of the `render_snippet` macro and more complicated examples.
+
+## Snippet Object
+
++ Data Structure: Attributes
+    + id (string) - Unique ID used to reference the snippet.
+    + content (object) - HTML, AMPHTML, and/or text content.
+        + html (string) - HTML content which will be used when the snippet is referenced from the HTML part of a template or inline content.
+        + text (string) - Plain text content which will be used when the snippet is referenced from the plain text part or headers of a template or inline content.
+        + amp_html (string) - AMPHTML content which will be used when the snippet is referenced from the AMPHTML part of a template or inline content.
+    + shared_with_subaccounts (boolean) - Whether this snippet can be used by subaccounts.
+
+## Create a Snippet [POST /snippets]
+
++ Data Structure
+    + id (required, string) - Unique ID used to reference the snippet.
+    + content (required, object) - HTML, AMPHTML, and/or text content. Substitution syntax is supported with the exception of the `render_dynamic_content()` and `render_snippet()` macro calls. In other words, snippets may not reference other snippets.
+        + html (string) - HTML content which will be used when the snippet is referenced from the HTML part of a template or inline content. Maximum length - 100KB.
+        + text (string) - Plain text content which will be used when the snippet is referenced from the plain text part or headers of a template or inline content. Maximum length - 100KB.
+        + amp_html (string) - AMPHTML content which will be used when the snippet is referenced from the AMPHTML part of a template or inline content. Maximum length - 100KB.
+    + shared_with_subaccounts (required, boolean) - Whether this snippet can be used by subaccounts.
+        + Default: false
+
+
++ Request (application/json)
+
+    + Body
+
+        ```
+        {
+            "id" : "header",
+            "content": {
+                "html": "<b>Our cool header snippet for {{name}}</b>"
+            }
+        }
+        ```
+
++ Response 200 (application/json)
+
+        {
+          "results": {
+            "id": "header"
+          }
+        }
+
++ Response 422 (application/json)
```

Comment:
I don't think it's worth documenting but most responses currently are 400, in fact all to date are